### DISCUSSION TITLE
WAVESHARE ESP32S3-GEEK TF card  reader not working

### DISCUSSION THREAD

**Discussion by TV-Guy** [BitMaker-hub/NerdMiner_v2#405] [Ideas]
So, the only way to change the password is to enter it on the TF card config.json file. This is needed to expand the number of SOLO pools we could use or to try to solo mine other SHA256 coins.
But the nerdminer software won't read the TF card.
Has anyone else tried to load from the TF card on these?
I can't get mine to read the file (yes, formatted Fat32)
Thanks.

**Comment by jsy-rg1570** [comment]
Dear,
I haven't implemented the Esp32 S3 Geek with sdcard. It would be good to add another entry in platformio.ini to have 2 firmwares with and without sdcard.

**Reply by jsy-rg1570** [reply]
I could try to look at this task soon.

**Comment by TV-Guy** [comment]
Thanks for looking at this for me.  I believe it would be worth implementing. Mike MavilleOn May 16, 2024, at 6:16 PM, Julio MATARRANZ ***@***.***> wrote:﻿
Dear,
I haven't implemented the Esp32 S3 Geek with sdcard. It would be good to add another entry in platformio.ini to have 2 firmwares with and without sdcard.

**Comment by jsy-rg1570** [comment]
Dear, 
I tried with this code https://github.com/julio-matarranz/NerdMiner_v2/tree/feat/s3-geek-sd but I got "The physical drive cannot work" maybe it's because of my sdcard. Feel free to try ;)

**Comment by TV-Guy** [comment]
Sorry but I'm new to all this.  I tried many many different things with my
VSC/platform.io and I'm afraid it's really messed up.
I can try compiled files (BIN) and I think my Aurduino IE still works but
the VSC is just not working and/or I don't know how to use it.
I did try the bin files that are in the directory but I had no result and I
suspect they are just the current files.
I'd like to help but you'll have to dumb it down for me.
Thanks
Mike

**Comment by jsy-rg1570** [comment]
https://github.com/julio-matarranz/NerdMiner_v2/releases/tag/nerdminer-prerelease-V1.6.3 you can try NerminerV2-S3-GEEK-SD_ bin files

**Comment by TV-Guy** [comment]
Ugh.   My VSC  has me loading rtosconfig.h and other files that should be included…I’ve been at this for hours and can’t compile.It keeps asking me to add other files not listed as errors until I add a missing file, then it asks for a new different file.  I clearly don’t know what I’m doing.  Mike MavilleOn May 16, 2024, at 6:21 PM, Julio MATARRANZ ***@***.***> wrote:﻿
I could try to look at this task soon.

**Comment by TV-Guy** [comment]
Geek-SD read the config file OK, started hashing fine but the screen was
blank.
I used the Boot_app0 files from the NM S3 geek files since only 3 files
were provided.

Thanks for all your effort on this.

Mike

**Comment by TV-Guy** [comment]
Hi Julio,
You are very close on this, the TF card is read and updated on the ESP32-S3-geek.
Unfortunately, the screen remains blank, not showing anything.
Thanks.

**Reply by jsy-rg1570** [reply]
> Hi Julio,
> 
> You are very close on this, the TF card is read and updated on the ESP32-S3-geek.
> 
> Unfortunately, the screen remains blank, not showing anything.
> 
> Thanks. 

Dear,
I will try something about that next week.
For now I only have my old Asus UX305 laptop which isn't really efficient and pretty boring to compile and test firmwares 😅

**Comment by TV-Guy** [comment]
Ok. Thanks for the reply.  Mike MavilleOn May 24, 2024, at 11:34 AM, Julio MATARRANZ ***@***.***> wrote:﻿

Hi Julio,
You are very close on this, the TF card is read and updated on the ESP32-S3-geek.
Unfortunately, the screen remains blank, not showing anything.
Thanks.

Dear,
I will try something about that next week.
For now I only have my old Asus UX305 laptop which isn't really efficient and pretty boring to compile and test firmwares 😅

**Reply by jsy-rg1570** [reply]
> Ok. Thanks for the reply.  Mike MavilleOn May 24, 2024, at 11:34 AM, Julio MATARRANZ ***@***.***> wrote:﻿ Hi Julio, You are very close on this, the TF card is read and updated on the ESP32-S3-geek. Unfortunately, the screen remains blank, not showing anything. Thanks. Dear, I will try something about that next week. For now I only have my old Asus UX305 laptop which isn't really efficient and pretty boring to compile and test firmwares 😅 —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

I made PR #414, this version is working with screen and sd card at same time.
Sd card is not necessary after init and it can work without sd card. ;)

**Comment by TV-Guy** [comment]
Thanks.
I tried for several hours to get my Platformio to compile but it's still
missing alot.
I'll test this if you can compile for me.
Sorry I'm not more of a help.
Thanks,
Mike
